sw/airborne/firmwares/rotorcraft/oneloop/oneloop_andi.c

```diff
@@ -0,0 +1,1524 @@
+    // First time engaging NAV Mode or Navigation Functions (e.g. Oval) requires some initialization
+    if(check_1st_nav){ 
+      // Mark that NAV Mode has been engaged
+      check_1st_nav  = false;
+      // Reset desired heading to current heading
+      psi_des_rad = eulers_zxy.psi; 
+    }
+    // Register Arttitude Setpoints from previous loop
+    float att_des[3] = {eulers_zxy_des.phi, eulers_zxy_des.theta, psi_des_rad};
+    // Generate Reference signals for positioning using RM
+    if (rm_order_h == 3){
+      rm_3rd_pos(dt_1l, pos_ref, pos_d_ref, pos_2d_ref, pos_3d_ref, nav_target, k1_pos_rm, k2_pos_rm, k3_pos_rm, max_v_nav, max_a_nav, max_j_nav, 2);    
+    } else if (rm_order_h == 2){
+      float_vect_copy(pos_ref, pos_1l,2);
+      rm_2nd_pos(dt_1l, pos_d_ref, pos_2d_ref, pos_3d_ref, nav_target, k2_pos_rm, k3_pos_rm, max_a_nav, max_j_nav, 2);   
+    } else if (rm_order_h == 1){
+      float_vect_copy(pos_ref, pos_1l,2);
+      float_vect_copy(pos_d_ref, pos_d,2);
+      rm_1st_pos(dt_1l, pos_2d_ref, pos_3d_ref, nav_target, k3_pos_rm, max_j_nav, 2);   
+    }
+
+    // The RM functions want an array as input. Create a single entry array and write the vertical guidance entries. 
+    float single_value_ref[1]        = {pos_ref[2]};
+    float single_value_d_ref[1]      = {pos_d_ref[2]};
+    float single_value_2d_ref[1]     = {pos_2d_ref[2]};
+    float single_value_3d_ref[1]     = {pos_3d_ref[2]};
+    float single_value_nav_target[1] = {nav_target[2]};
+    float single_value_k1_rm[1]      = {k1_pos_rm[2]};
+    float single_value_k2_rm[1]      = {k2_pos_rm[2]};
+    float single_value_k3_rm[1]      = {k3_pos_rm[2]};
+
+    if (rm_order_v == 3){
+      rm_3rd_pos(dt_1l, single_value_ref, single_value_d_ref, single_value_2d_ref, single_value_3d_ref, single_value_nav_target, single_value_k1_rm, single_value_k2_rm, single_value_k3_rm, max_v_nav, max_a_nav, max_j_nav, 1);    
+      pos_ref[2]    = single_value_ref[0];
+      pos_d_ref[2]  = single_value_d_ref[0];
+      pos_2d_ref[2] = single_value_2d_ref[0];
+      pos_3d_ref[2] = single_value_3d_ref[0];
+    } else if (rm_order_v == 2){
+      rm_2nd_pos(dt_1l, single_value_d_ref, single_value_2d_ref, single_value_3d_ref, single_value_nav_target, single_value_k2_rm, single_value_k3_rm, max_a_nav, max_j_nav, 1);   
+      pos_ref[2]    = pos_1l[2];
+      pos_d_ref[2]  = single_value_d_ref[0];
+      pos_2d_ref[2] = single_value_2d_ref[0];
+      pos_3d_ref[2] = single_value_3d_ref[0];
+    } else if (rm_order_v == 1){
+      rm_1st_pos(dt_1l, single_value_2d_ref, single_value_3d_ref, single_value_nav_target, single_value_k3_rm, max_j_nav, 1); 
+      pos_ref[2]    = pos_1l[2];
+      pos_d_ref[2]  = pos_d[2];
+      pos_2d_ref[2] = single_value_2d_ref[0];
+      pos_3d_ref[2] = single_value_3d_ref[0];  
+    }    
+    // Generate Reference signals for attitude using RM
+    // FIX ME ow not defined anymore without oval
```

Comment:
Can this be fixed?

Reply:
The way i left this comment was not clear. This boolean can be useful in the future if we would like to specify heading rates and accelerations in a particular NAV function. In the past I have used this for correct heading tracking in the oval but can be left there for when we will have heading references for NAV functions.  I updated the ccomment.